llvm/lib/Target/AIE/AIEMachineScheduler.cpp

```diff
@@ -1148,11 +1163,41 @@ void llvm::AIEPostRASchedStrategy::buildGraph(ScheduleDAGMI &DAG, AAResults *AA,
 
   auto &BS = InterBlock.getBlockState(CurMBB);
   const auto &Region = BS.getCurrentRegion();
-  for (auto *I : Region) {
-    DAG.initSUnit(*I);
+  int NCopies = 1;
+  if (int II = BS.FixPoint.II) {
+    assert(BS.Kind == BlockType::Loop);
+    assert(BS.getRegions().size() == 1);
+    // Try to wrap the linear schedule within II.
+    // We virtually unroll the body by the stagecount, computed from rounding
+    // up the length divided by II.
+    NCopies = (BS.getScheduleLength() + II - 1) / II;
```

Comment:
The "standard" schedule has been achieved using different scheduling techniques. Can we guarantee there will be enough copies if the linear schedule used for pipelining is different?

Reply:
Good point. I could make it safe (for now) by rejecting NStages > NCopies. What do you think?